server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -0,0 +1,21 @@
+package org.apache.james.jmap.http
+
+import javax.inject.Inject
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import reactor.core.publisher.Mono
+import reactor.core.scala.publisher.SFlux
+import reactor.netty.http.server.HttpServerRequest
+import scala.compat.java8.StreamConverters._
+class BasicAuthenticationStrategy @Inject() (val basicAuthTokenManager: BasicAuthenticationTokenManager,
+                                             val mailboxManager: MailboxManager) extends AuthenticationStrategy {
+
+  override def createMailboxSession(httpRequest: HttpServerRequest): Mono[MailboxSession] = {
+    SFlux.fromStream(() => authHeaders(httpRequest).toScala[Stream])
+      .map(basicAuthToken => basicAuthTokenManager.removeBasicAuthenticationPrefix(basicAuthToken))
+      .filter(basicAuthToken => basicAuthTokenManager.isValid(basicAuthToken))
+      .map(basicAuthToken => basicAuthTokenManager.retrieveUserNameFromCredential(basicAuthToken))
+      .map(username => mailboxManager.createSystemSession(username))
```

Comment:
IMO the pipeline should be:

 - Extract authentication header
 - Keep `Basic` ones
 - Retrieve associated `UserCredentials`
 - Validate the `UserCredentials` are valid
 - create the user session

removeBasicAuthenticationPrefix is an implementation details that should be down in the method call stack

isValid should not be applied on raw data as we end up doing the user request parsing two times.